src/test/java/com/juniorstart/juniorstart/service/ProjectServiceTest.java

```diff
@@ -0,0 +1,172 @@
+    TechnologiesRepository technologyRepository = Mockito.mock(TechnologiesRepository.class);
+
+    ProjectService projectService = new ProjectService(userDao,projectRepository, technologyRepository);
+
+    Instant createdAt;
+    Project project;
+    Technologies technology;
+    UserSummary userSummary;
+    User user;
+    UserSummary mentor;
+    Pageable pageable;
+    List<Project> projectList;
+    List<User> users;
+    UserPrincipal userPrincipal;
+    ProjectRequest projectRequest;
+    Set<Project> projectSet;
+    HashMap<Long, User> creatorMap;
+    Page<Project> page;
+
+    @Before
+    public void initialize() throws Exception {
+        technology = Technologies.builder().technologyType(TechnologyType.FRONTEND).description("Some description").id(1l).title("Some title").build();
+        createdAt = new SimpleDateFormat("yyyy-MM-dd").parse("2020-12-31").toInstant();
+        users = new ArrayList<>();
+        user = new User();
+        user.setPrivateId(UUID.randomUUID());
+        user.setPublicId(21l);
+        user.setEmailVerified(true);
+        user.setName("Some name");
+        user.setProvider(AuthProvider.local);
+        user.setPassword("Somepassword");
+        user.setAge(12);
+        user.setHiddenFromSearch(true);
+        user.setImageUrl("SomeImageUrl");
+        user.setUserStatus(UserStatus.LOOKING_FOR_A_JOB);
+        user.setEmail("some@example.com");
+        project = new Project();
+        project.setBody("Some body");
+        project.setDescription("Some description");
+        project.setNumberOfSeats(12l);
+        project.setTitle("Some title");
+        project.setCreatedAt(createdAt);
+        project.setCreatedBy(user.getPublicId());
+        project.setCreatedAt(createdAt);
+        project.setUpdatedBy(user.getPublicId());
+        project.setId(12l);
+        project.setName("Some Easy name");
+        pageable = PageRequest.of(0, 10, Sort.Direction.DESC, "createdAt");
+        projectList = new ArrayList<>();
+        projectList.add(project);
+        page = MockUtil.createMockPage(projectList);
+        userPrincipal = UserPrincipal.create(user);
+        projectRequest = new ProjectRequest();
+        projectRequest.setBody(project.getBody());
+        projectRequest.setDescription(project.getDescription());
+        projectRequest.setName(project.getName());
+        projectRequest.setNumberOfSeats(project.getNumberOfSeats());
+        projectRequest.setRepository(project.getRepository());
+        projectRequest.setTitle(project.getTitle());
+        technology.setCreatedAt(createdAt);
+        technology.setUpdatedAt(createdAt);
+        technology.setCreatedBy(user.getPublicId());
+        technology.setUpdatedBy(user.getPublicId());
+        project.getTechnologies().add(technology);
+        project.setRepository("Some Repo");
+        project.setUpdatedAt(createdAt);
+        projectSet = new HashSet<>();
+        projectSet.add(project);
```

Comment:
Te sety to do zewnętrznych funkcji chyba dobrze by było rozrzucić. Np. initializeUser, initializeProject, itd.